### ISSUE TITLE
paginate() (because of count()) EXTREMELY slow/inefficient (When using QueryCache).

### ISSUE THREAD

**Issue by EVILoptimist** [propelorm/Propel2#960]
Ran into a performance issue today when I needed to use the ->paginate() functionality on a table with 30k+ records on MySQL. For a simple query, it appears as though Propel chooses to perform the count on a derived table instead of simply selecting COUNT(*) instead of the field list.

How I would query MySQL in this case: 0.292sec
```sql
SELECT COUNT(*) FROM  `items` propelmatch4cnt
```

How PropelORM chooses to query MySQL: 58.187sec
```sql
SELECT COUNT(*) FROM (SELECT <<LENGTHY LIST OF FIELDS>> FROM `items`) propelmatch4cnt
```

Considering that the ->paginate() functionality exists almost exclusively for dealing with large recordsets, this seems to be extremely counterintuitive.

**Comment by EVILoptimist** [comment]
After looking into the Base classes, I can see that there is logic in place that is meant to execute a much faster query.

To clarify, this is the code being used that generated this query:
```php
$query = ItemQuery::create()->count();
```

**Comment by EVILoptimist** [comment]
Additionally the count() method of the PropelModelPager returns the getLimit() value instead of the value that it took 58 seconds to retrieve :)

```php
$items = ItemQuery::create()->paginate(1, 10);
$items->count(); // int(10)
```

**Comment by EVILoptimist** [comment]
OK. Looks like I found a simple solution after reviewing the way this is handled in other parts of the code.

Pull request submitted! :)

**Comment by marcj** [comment]
Fixed through the submitted PR #961, thanks!
